File: maze/Door.java
package Maze;

public class Door
{
	private Question question;
	private Boolean isLocked;
	
	public Door(Question question, Boolean isLocked)
	{
	    if(question == null)
        {
            throw new IllegalArgumentException("Question cannot be null!");
        }
	    if(isLocked == null) {
            throw new IllegalArgumentException("Passed in boolean cannot be null!");
        }
		this.question = question;
		this.isLocked = isLocked;
	}

	//Testing purposes
    @Override
    public String toString()
    {
        return (this.question.toString() + " " + this.isLocked.toString());
    }
}

File: EntertainmentMaze/maze/Door.cs
using EntertainmentMaze.Database;
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;
using System.Runtime.Serialization;

namespace EntertainmentMaze.maze
{
    [DataContract]
    public class Door
    {
        [DataMember]
        private Question Question { get; set; }
        [DataMember]
        private bool IsLocked { get; set; }
        [DataMember]
        private bool IsOpened { get; set; }
        [DataMember]
        private string QuestionString { get; set; }
        [DataMember]
        private string AnswerString { get; set; }

        public static Door CreateDoor() => new Door();
        public Question GetQuestion() => Question;
        public bool GetDoorStatus() => IsLocked;
        public bool GetDoorOpenedStatus() => IsOpened;
        public string GetQuestionString() => QuestionString;
        public string GetAnswerString() => AnswerString;

        public Door()
        {
            Question = GetRandomQuestion();
            QuestionString = Question.CurrentQuestion;
            AnswerString = Question.Answer;
            IsLocked = false;
        }

        private Question GetRandomQuestion()
        {
            Random rnd = new Random();
            List<Question> listOfQuestions = DatabaseListRetrieval.GetListOfQuestions();
            int r = rnd.Next(listOfQuestions.Count);
            Question question = listOfQuestions.ElementAt(r);

            return question;
        }



        public void OpenDoor()
        {
            IsOpened = true;
        }
        public void LockDoor()
        {
            IsLocked = true;
        }

        public void UnlockDoor()
        {
            IsLocked = false;
        }

        public string DisplayQuestion()
        {
            return ($"{Question.CurrentQuestion.ToString()}");
        }

        public string DisplayAnswer()
        {
            return ($"{Question.Answer.ToString()}");
        }
    }
}
